### DISCUSSION TITLE
VM370 NJE to NOS2.8.7

### DISCUSSION THREAD

**Discussion by tchandler48** [kej715/DtCyber#159] [Q&A]
I am trying in my home lab set up a connection from VM370 r6 NJE to NOS2.8.7.  I have a working VM370 RSCS connection as I have linked to other systems in the pass.  I can not find any understandable documentation on how to perform this.  I have the VM370 to NOS connection setup and the link exchanges TCP buffer information.  But the link never comes active.  Any guidance on documentation or howto would be most helpful.  Thank You

**Comment by drboone** [comment]
Pretty sure that unless you have Peter Coghlan's NJE patches, R6 RSCS doesn't do NJE.  But last I fiddled, the DtCyber/NOS setup does RJE anyway.

**Comment by tchandler48** [comment]
I have Peter's patches (level 7) and it is working well.  I have the vm370
trying to attach to NOS, it exchanges buffer
size requests, but can not get any further.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

Please provide more information. Did you install your NOS 2.8.7 system from
one of the read-to-run packages described here
<https://github.com/kej715/DtCyber/tree/main/NOS2.8.7#installopts>?  If so,
did you configure your RSCS node to NOS 2.8.7 by adding an njeNode
definition to the [NETWORK] section of your site.cfg file, as described here
<https://github.com/kej715/DtCyber/tree/main/NOS2.8.7#njeNode>? And did you
then run the NJE configuration script by entering the "njfc" command at the
DtCyber console, or by running the njf-configure.js script using node, as
in:

node njf-configure


If you did these things, then please provide a copy of your site.cfg file,
your cyber.ovl file, and your VM370CE system's config file. It might also
be useful to have a copy of your NDL definition file, your NJF
configuration file, and your LIDCMxx file.

Also, did you try using the NJF *K* display to see which nodes are
connected to NJF and what activity is occurring? Viewing its running
DAYFILE can be enlightening too (DAYFILE,NJF.).

Note that configuring NJE on NOS 2.8.7 manually, without the help of the
njf-configure.js script, is not for the faint-hearted. It requires
knowledge of CCP, NDL, and NJF. The NDL configuration file needs to be
updated to define a line for each NJE peer node, the NOS LIDCMxx file needs
to be updated to define PID's and LID's for NJE nodes, and the NJF
configuration file needs to be created/updated to define information about
the NJE network (in alignment with the LIDCMxx file). DtCyber's cyber.ovl
file also needs to be udpated to define TCP connection info for NJE peer
nodes. NJF configuration documentation is found in the NJEF installation
guide which is a file on the NJEF product distribution tape.

The njf-configure.js script automates the process of updating the NDL,
LIDCMxx, and NJF configuration files, and it also updates
DtCyber's cyber.ovl file to align with the NDL configuration.

cheers,
Kevin

**Comment by tchandler48** [comment]
Kevin,
Thank you for the hints.  Some I figured out last night, so new information
I need to test.
Short answer, to most of your questions is NO, so I have a lot of
learning/testing today
then I will report back to you in detail.

Thank you for all the effort you have put into this.  IT is GREAT.

/cheers
/tom c

**Comment by tchandler48** [comment]
Kevin,
Started testing based on your email.  I added nodeNJE to my site.cfg file:
njeNode=6PEXTA,RSCS,CMS,192.168.1.202:7900,NCCM01,,,
          6PEXTA     VM370 Node Name
          RSCS        Level 7 by Peter C
          CMS         Identifier
          192.168.1.202:7900   the ip/port that VM370 Listens on
           NCCM01   Gateway my NOS2.8.7 system   full-865

At the operator console, when I type njfc, it runs down to
          Compile NJF host configuration file......
 and dies and exits with status 1

BTW, If i try node njf-configure, it also dies.....

So, I think I will do a full install of full-865 and start fresh. I could
have
fat fingered something in all my testing, so I think starting fresh
would make the most sense.

Any suggestions, I am open.

Thank you for all your help.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

Starting fresh feels like a good idea, especially if you've been editing
various config files manually. However, another possible source of problems
might be the name you've assigned to your VM370CE node. The node name
"6PEXTA" might be problematic. In particular, the NJF configuration file
compiler might not accept a node name beginning with a digit. It might
require node names to begin with alphabetic characters. I'm not absolutely
sure about that, but I'd be wary of defining node names that don't begin
with letters. Historically, no nodes in the BITNet network had names
beginning with digits, and there are currently no nodes in the HNET network
with names beginning with digits.

cheers,
Kevin

**Comment by tchandler48** [comment]
Sir,
Clean install of full-865.  IPL's and run's great!

I have in my site.cfg the following:
[CMRDECK]
MID=23.
[NETWORK]
hostID=NCCM01
njeNode=VMEXTA,RSCS,CMS,192.168.1.202:7900,NCCM01

The above is all I have in site.cfg.......

I run njfc and it dies on the COMPHCF failed.  It has to be something in my
site.cfg.
I have studied the manual and the layout of the command and the example of
the command
I can not match it up.  I am missing something I am sure.

Any suggestions are REALLY welcome.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

When did you create/update the site.cfg file to define the machine id (MID)
as 23? If you created/updated site.cfg to define MID=23 -after- running the
initial install script to install full-865, then simply running "ncfg" to
update the NJF configuration would be insufficient. The machine ID
influences many aspects of the overall machine configuration. "ncfg"
updates the NJE configuration only.

If.you set/change MID in the [CMRDECK] section of site.cfg -after-
performing an initial install, then you need to run the general
reconfiguration script, "rcfg", to ensure that all aspects of the machine
configuration dependent upon the machine ID are updated. The "rcfg" script
will call the "ncfg" script automatically, as needed, along with calling
other scripts that are also dependent upon the machine ID.

Note also that the various re/configuration scripts run batch jobs to do
their work, and these batch jobs generate output and log information that
is printed on the system's line printer. Line printer output is written to
the file named LP5xx_C12_E5. When things go wrong while running
re/configuration scripts, viewing this file will tend to reveal important
clues.

Try running the general reconfiguration script, "rcfg".

cheers,
Kevin

**Comment by tchandler48** [comment]
Kevin,
Thank you for all the information.  Recap,
Fresh install full-865
No njeNode entry in site.cfg

Ran rcfg,system ipl and appears to be acting normally.

Then, I modified site.cfg adding njenode information.
Ran rcfg - OK
When I tried to un njfc, It does not compile.

So, something about the njenode entry I am using, and/or missing additional
entries in other config files.  At this point, I don't know.

I don't how much time you want to spend on this, you have do a LOT....Thank
you for your help.

I am going to read and try to understand it better.  I would like to get
this going, but at
some point, you have to step back.

One question, Do I need to put an entry in my topology file for 6PEXTA
entry and also in the njeNode(site.cfg)

Again, Thank you very much for all your help.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

Are you sure that "rcfg" ran successfully to completion? I just performed
the following:

   1. node install rtr nos287-full-865
   2. Created site.cfg with MID=23 and njeNode=6PEXTA,RSCS,CMS,
   192.168.1.202:7900,NCCM01, as originally defined by you.
   3. Ran "rcfg"

The reconfiguration script (rcfg) failed because the sub-script
njf-configure.js failed.. Review of the job output written to LP5xx_C12_E5
revealed that the NJF configuration file compiler, HCFP, failed because the
LID specified for node 6PEXTA in site.cfg, CMS, was a duplicate. This LID
"CMS" is already assigned to the HNET node NCCCMS, as defined in the
nje-topology.json.

I changed the LID in site.cfg from "CMS" to "6PE" and reran the
reconfigure.js script. It succeeded after making the change.

So, the cause of the problem is that the LID "CMS" is already taken, and
you need to choose a LID that isn't defined already in the
nje-topology.json file. I chose "6PE", and this resolved the problem. Note
that this also shows that the NJF configuration file compiler actually
-does- accept "6PEEXTA" as an NJE node name.

cheers,
Kevin

**Comment by tchandler48** [comment]
Kevin,
Your email about the CMS changing to 6PE fixed that problem.  THANK YOU
VERY MUCH.

Looking in the LP file, I found the following, which I don't understand.
THese are errors in the file:

htfconf not found
rexconf not found
smiconf not found.

and job ICF aborts.

How would I go about fixing this?.

/cheers
/tom c

**Comment by tchandler48** [comment]
Kevin,
Think I got it.................................
More testing tonight and tomorrow I send you results.

Short answer,  READ THE MANUAL AND DO WHAT IT STATES, NOT WHAT YOU
THINK.............

Until tomorrow.....
/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

Regarding:

htfconf not found
rexconf not found
smiconf not found.


There's nothing to be fixed here. These are informative messages indicating
that these configuration files have not been defined. These configuration
files are optional. When they're not defined (the normal case for the
ready-to-run packages), the respective applications use default
configuration parameters which are sufficient for most purposes.

Regarding:

job ICF aborts.


ICF is the Interactive Cray Facility. ICF is dependent upon the CRS (Cray
Station) subsystem. The ready-to-run packages do not activate CRS, so ICF
will not run successfully by default. In other words, it is normal and
expected for the ICF job to abort in ordinary ready-to-run configurations.
However, if you have also installed Andras Tantos' Cray X-MP simulator,
then you can add a "crayStation" definition to the [NETWORK] section of
your site.cfg file and execute the "rcfg" command to reconfigure your
system to compile and activate the Cray Station software. This will enable
your system to activate the CRS subsystem which will them connect the Cray
X-MP system, and then the ICF job will start successfully. ICF enables you
to log into the Cray X-MP system interactively via the NOS 2.8.7 system.

cheers,
Kevin

**Comment by tchandler48** [comment]
Kevin,

Still working on it, close but not yet....Couple of ????

I have using Linux Ubuntu22.04 Desktop
hercules 3.13  _VM end
rscs level 7 (Peter C mods) VM_end

I wonder what version of rscs the current vm370 users are using?

Does the dtcyber work ok under Linux???

THank you for all your help

/cheers
/tom c

**Comment by kej715** [comment]
Hi Peter,

Please clarify what you mean by "current vm370 users". If you're referring
to users of VM370CE, then all users of that distribution will be using a
version of RSCS that is compatible with Peter's mods (i.e., probably RSCS
Level 7).  RSCS at the Nostalgic Computing Center runs on VM370CE 1.2, and
it has Peter's mods applied.

Yes, DtCyber works OK under Linux. For example, the Cyber 175 running NOS
2.8.7, and the Cyber 173 running NOS 1.3 at the Nostalgic Computing Center
are running under Linux on Raspberry Pi systems, and previously the Cyber
175 ran under Linux on a Mac Mini (the Mac Mini was running Linux natively
instead of MacOS). Many members of the DtCyber community (maybe even most?)
run DtCyber on Linux.

cheers,
Kevin

**Comment by tchandler48** [comment]
Sir,
I think the above message was for me, if not it had some great information
anyway.

question:  You stated in that message that DtCyber was running on
vm370ce_1.2.

I am using 6PExta not CE and Hercules 3.13 not SDL.  Do you think that could
impact what I am trying to achieve?.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

First, apologies for the previous salutation error.

I'm not familiar with "6PExta" and quick google searches didn't reveal any
information about a Hercules or VM/CMS distribution by that name. Getting
RSCS to run and communicate successfully does require specific features in
Hercules, and I'm not sure whether Hercules 3.13 provides them. The VM370CE
distributions include RSCS, Peter's mods are compatible, and the
combination is known to work well. Beyond that, VM370CE is well-curated and
contains many other features including a variety of programming languages
and utilities. It also includes the MAIL utility that I contributed, which
enables you to send/receive email via RSCS (i.e., BITNet-style email). I
recommend using the latest release of VM370CE on Hercules SDL. If you
provide me with a private email address, I can provide step-by-step
instructions for applying Peter's mods against it and getting BITNet-style
email running.

cheers,
Kevin

**Comment by tchandler48** [comment]
Thank you for the response.  6PExt was a release put out by Prof. Ferland
and I really liked it.
I use it for the NICOF support, which works great.  TCPIP on VM370........

I have installed VM370CE_1._1_2 under Hercules 3.13.  If you have
instructions on RSCS install under
VMCE, It would be very helpful.

Thank you for all your help and effort and time spent.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

Here is a DropBox link to instructions for applying Peter Coghlan's NJE
mods to VM370CE:


https://www.dropbox.com/scl/fi/kmd02gt8hxo6ldoaj7ekl/nje4vm370ce-memo.txt?rlkey=9zz3h2ihiw8e14l8o5kqi9bxw&st=7mgzq4pc&dl=0

Note, however, that while VM370CE 1.1.2 might run on Hercules 3.13, it's
not clear that RSCS will operate well on 3.13. RSCS has specific
requirements for data communication (e.g., BSC emulation) that 3.13 might
not support as expected. Apparently, Hercules 3.13 is about 10 years older
than the latest Hercules SDL (aka 4.7).

cheers,
Kevin

**Comment by tchandler48** [comment]
Downloaded the instructions.  Next I downloaded fresh copy of VM370CE_1_1_2

Went thru steps 1 to 5, no problem.  However when I execute step 6
re-logon of MAINT,
the system crashes with:
CMSFRE161T INVALID DMSFRET CALL FROM F87AE8,ERROR NUMBER 5.
DMSFRE164T LOW-CORE NUCLEUS STORAGE POINTERS DESTROYED (INTERNAL ERROR CODE
5) RE-IPL CMS

When I try to re-ipl cms, it
LOOPS WITH THE TWO ABOVE ERROR MESSAGES.

The only difference is I am using spinhawk, not sdl.  I guess I should gen
up a SDL version and run the
same test as above to see if I get different results or the same.

/cheers
/tom c

**Comment by tchandler48** [comment]
Well, add this to the don't do it file.......................

Then system netid changes as outlined in your notes DOES NOT work using
Hercules Spinhawk 3.13
However
on SDL 4.0 it works perfectly.  So I am running SDL for my
testing...............

Now on to step 7 and beyond.........................

/cheers
/tom c

**Comment by tchandler48** [comment]
Kevin,
the rscsupdt.vmarc file I have has a bad header.  I tried the dropbox file,
but it is no longer there.
Do you have a copy of the nje4rscs file, as the one I have has an error.

thank you
tom c

**Comment by kej715** [comment]
Hi Tom,

The nje4rscs.vmarc file was a copy of Peter Coghlan's mods for RSCS. It was
not appropriate to make it public on DropBox, and it was removed at Peter's
request. If you need a fresh copy, you will need to obtain it directly from
Peter.

cheers,
Kevin

**Comment by tchandler48** [comment]
Sir,
Last evening Peter sent me a new updated tape, No Problem.

I do have a question on the instructions you provided on how to install/gen
rscs on VM370CE.
Step11.  LAXLINES.  I can not find a sample copy book for this.  Do I need
to create a NEW
file name LAXLINES COPY and put the GENLINE LINE=070 inside of it?

So far up to this point, all is well.

BTW, It appears that the SYSTEM NETID change will NOT work under Spinhawk.
I install
SDL 4.0 and it works fine.

/cheers
/tom c

**Comment by tchandler48** [comment]
Kevin,
Thank you for ALL YOUR HELP.  I am moving on from trying to link VM370 to
NOS.  I am missing
something(s) and just have run out of get up and go.........

I am still playing  around learning, NOS is great system.

Hope things are going well.

/cheers
/tom c

**Comment by kej715** [comment]
Hi Tom,

If LAXLINES COPY A does not exist at step 11 of the instructions, you will
need to create it. Are you sure that you executed step 9 where "SETUP RSCS"
is executed and minidisk 59F is attached to A? If you didn't execute that
step, then you will not find LAXLINES COPY on MAINT's default minidisk A.

cheers,
Kevin
